flake.nix

```diff
@@ -0,0 +1,140 @@
+{
+  description = "Nix flake for zkLLVM";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+    nix-3rdparty = {
+      url = "github:NilFoundation/nix-3rdparty";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+    nil-crypto3 = {
+      url = "https://github.com/NilFoundation/crypto3";
+      type = "git";
+      submodules = true;
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+      };
+    };
+    nil-zkllvm-blueprint = {
+      url = "https://github.com/NilFoundation/zkllvm-blueprint";
+      type = "git";
+      submodules = true;
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        nil_crypto3.follows = "nil-crypto3";
+      };
+    };
+  };
+
+  outputs =
+    { self
+    , nixpkgs
+    , flake-utils
+    , nix-3rdparty
+    , nil-crypto3
+    , nil-zkllvm-blueprint
+    }:
+    flake-utils.lib.eachDefaultSystem (system:
+    let
+      pkgs = import nixpkgs {
+        overlays = [ nix-3rdparty.overlays.${system}.default ];
+        inherit system;
+      };
+      crypto3 = nil-crypto3.packages.${system}.default;
+      blueprint = nil-zkllvm-blueprint.packages.${system}.default;
+
+      # Default env will bring us GCC 13 as default compiler
+      stdenv = pkgs.stdenv;
+
+      defaultNativeBuildInputs = [
+        pkgs.cmake
+        pkgs.ninja
+        pkgs.python3
+        pkgs.git
+      ];
+
+      defaultBuildInputs = [
+        # Default nixpkgs packages
+        pkgs.boost
+        # Repo dependencies
+        crypto3
+        blueprint
+      ];
+
+      defaultDevTools = [
+        pkgs.clang_17 # clang-format and clang-tidy
+      ];
+
+
+      defaultCmakeFlags = [
+        "-DCMAKE_CXX_STANDARD=17"
+        "-DBUILD_SHARED_LIBS=TRUE"
+        "-DZKLLVM_VERSION=0.1.18"
+      ];
+
+      releaseBuild = stdenv.mkDerivation {
```

Comment:
Please move this derivation to a separate file, same as derivations in nix-3rdparty

Reply:
In fact, I guess it is so because we were looking on [zkEVM-framework flake](https://github.com/NilFoundation/zkEVM-framework/blob/master/flake.nix). Here is [the issue](https://github.com/NilFoundation/zkEVM-framework/issues/66) which addresses the same problem: we need to refactor flakes.

Reply:
true, I've taken zkevm flake as basis

Reply:
Done